constructiveness_toxicity_crowdsource/common/data_input_output_functions.py

```diff
@@ -8,13 +8,29 @@ def get_full_annotation_csv(data_path, batch):
     :data_path: (str) the path of the directory containing annotated data CSVs
     :batch: (int) annotation batch number
     :return: (str) the path containing the full annotation CSV
+    
+    Description: Goven the data path and the batch number, extract and return the 
+    path of the CSV containing full data annotations. The input data path may contain 
+    multiple data CSVs. We want to extract the one following the form: 
+    batch[\d]_f[\d]+.csv
+    
+    Example:
+    >>> batch = 8    
+    >>> data_path = '../CF_output/constructiveness/batch8/'
+    >>> get_full_annotation_csv(data_path, batch)
+    >>> '../CF_output/constructiveness/batch8/batch8_f1285429.csv'
     '''
     
     input_csv_files = [f for f in glob.glob(data_path + 'batch' + str(batch) + '_f*.csv') 
                                                if ntpath.basename(f).count('_') == 1]
-    if len(input_csv_files) > 1:
-        print('Please review your input dir. There seem to be more than one input files. ')
-        sys.exit(0)
-    else:
+    try: 
         return input_csv_files[0]
-    
\ No newline at end of file
+    except: 
+        print('Exception: Please review your input dir. Full annotation file not found under the following dir: \n', data_path)
```

Comment:
This is not what I meant. I think the best thing would be to just use:

assert len(input_csv_files) == 1, "Your input directory should have exactly one input file."

That way you raise an exception if the assertion is false. We don't want to catch the exception, we just want the program to fail. You also don't need sys.exit(0). The reason we don't rely on print statements is that those are not always printed to a location where we can see them (e.g. if this is running on the cloud). And just exiting means we may not know whether the program succeeded or failed.

Reply:
Makes sense. Thanks! Just pushed the changes.